aml/src/resource.rs

```diff
@@ -1,112 +1,293 @@
 use crate::{value::AmlValue, AmlError};
+use alloc::vec::Vec;
 use bit_field::BitField;
 use byteorder::{ByteOrder, LittleEndian};
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub enum Resource {
     Irq(IrqDescriptor),
+    WordAddressSpace(AddressSpaceDescriptor),
+    MemoryRange(MemoryRangeDescriptor),
+    IOPort(IOPortDescriptor),
+    Dma(DMADescriptor)
 }
 
-/// Parse a `ResourceDescriptor`. Returns `AmlError::IncompatibleValueConversion` if the passed value is not a
+
+/// Parse a `ResourceDescriptor` into a list of resources. Returns `AmlError::IncompatibleValueConversion` if the passed value is not a
 /// `Buffer`.
-pub(crate) fn resource_descriptor(descriptor: &AmlValue) -> Result<Resource, AmlError> {
+pub fn resource_descriptor_list(descriptor: &AmlValue) -> Result<Vec<Resource>, AmlError> {
```

Comment:
The diff here is wonky. I'm not removing `resource_descriptor`. It still exists; I added a new `resource_descriptor_list` that parses a list of them.

I made it `pub` here. But `resource_descriptor` is still `pub(crate)`. Is it okay to make it public?